frontend/packages/dev-console/src/components/import/SamplesCatalog.tsx

```diff
@@ -11,6 +12,12 @@ type SampleCatalogProps = RouteComponentProps<{ ns?: string }>;
 const SampleCatalog: React.FC<SampleCatalogProps> = ({ match }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
+  const params = new URLSearchParams(window.location.search);
+  const sampleType = params.get('sampleType');
+  const labelFilter: Record<string, string> = {
+    label: 'sample-type',
+    value: sampleType,
+  };
```

Comment:
In my previous comment, I mixed two options, a `sampleType` query parameter is one option. The Record is an option for a more generic solution where the filter is something like `?labelFilter=serverless=true`

So this code currently mixes both, let us focus on one solution or the other. At the moment a sampleType is fine.

```suggestion
  const sampleType = params.get('sampleType');
```

Reply:
We need both since typeLabel is translated string in other samples